tests/test_deploy_github_pages.py

```diff
@@ -0,0 +1,277 @@
+                                        capture_output=True, text=True, check=True)
+        assert doc_dir_source_one_exists.returncode == 0
+        doc_dir_source_two_exists = run(["git", "ls-tree", "-d", f"origin/{target_branch}:{source_branch_two}"],
+                                        capture_output=True, text=True, check=True)
+        assert doc_dir_source_two_exists.returncode == 0
+        remove_branch(target_branch)
+
+
+def test_no_new_push_and_commit_if_no_changes():
+    with TemporaryDirectory() as tempdir:
+        os.chdir(tempdir)
+        user_name, user_access_token = setup_test_repo()
+        doc_dir = run(["git", "rev-parse", "--show-toplevel"], capture_output=True, text=True, check=True)
+        os.chdir(f"{doc_dir.stdout[:-1]}/doc")
+        source_branch = "5-add-tests"
+        run(["git", "checkout", source_branch], check=True)
+        target_branch = "test-docu-new-branch"
+        remove_branch(target_branch)
+
+        cwd = os.getcwd()
+        deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                                 "--push_origin", "origin",
+                                                 "--push_enabled", "push",
+                                                 "--source_branch", source_branch,
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+        current_commit_id = run(["git", "ls-remote", f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git", target_branch], capture_output=True, check=True)
+        commit_id_old = current_commit_id.stdout
+
+        deploy_github_pages.deploy_github_pages(["--target_branch", target_branch,
+                                                 "--push_origin", "origin",
+                                                 "--push_enabled", "push",
+                                                 "--source_branch", source_branch,
+                                                 "--source_dir", cwd,
+                                                 "--module_path", ["../test_package", "../another_test_package"]])
+        current_commit_id = run(["git", "ls-remote", f"https://{user_access_token}@github.com/exasol/sphinx-github-pages-generator-test.git", target_branch], capture_output=True, check=True)
+        commit_id_new = current_commit_id.stdout
+
+        # the docu files where not updated, so no new commit should be pushed to the remote
+        assert commit_id_old != 0 #todo also in other tests
+        assert commit_id_old == commit_id_new
+        assert not commit_id_new == ""
+        remove_branch(target_branch)
+
+
+def test_for_existence_of_docu_files():
+    # generate files locally in test, and with generator, look at diff between local files and remote files?
```

Comment:
Hm the test name is a bit confusing. Actually you don't check if there was any html generated, but you verify if the output of deploy_github_pages.deploy_github_pages() changes anything between local and remote branch

Reply:
Changed